task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,265 @@
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --event='error|warning'
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as Snyk only provides an URI to embed the context
+            csgrep --mode=json --prepend-path-prefix="${SOURCE_CODE_DIR}"/ raw_sast_unicode_check_out.txt \
+                | csgrep "${CSGERP_OPTS[@]}" \
+                | csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ \
+                > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        filter_kfp()
+        {
+            # We check if the KFP_GIT_URL variable is set to apply the filters or not
+            if [[ -z "${KFP_GIT_URL}" ]]; then
+                mv processed_sast_unicode_check_out.json sast_unicode_check_out.json
+                echo "Error: KFP_GIT_URL variable not defined. False positives won't be filtered" >&2
+                return 0
+            else
+                echo "Message: Filtering false positives in results files using csfilter-kfp..." >&2
+                CMD=(
+                    csfilter-kfp
+                    --verbose 
+                    --kfp-git-url="${KFP_GIT_URL}"
+                )
+
+                if [[ -n "${PROJECT_NVR}" ]] && [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                    CMD+=(
+                        --project-nvr="${PROJECT_NVR}"
+                        --record-excluded=raw-excluded-findings.json
+                    )
+                fi
+
+                "${CMD[@]}" processed_sast_unicode_check_out.json  > sast_unicode_check_out.json
+                status=$?
+                if [ "$status" -ne 0 ]; then
+                    echo "Error: failed to filter known false positives" >&2
+                    return 1
+                else
+                    echo "Message: Succeed to filter known false positives" >&2
+                fi
+
+                if [ -f "raw-excluded-findings.json" ]; then
+                    csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ raw-excluded-findings.json > excluded-findings.json
+                fi
+            fi
+        }
+
+        show_output()
```

Comment:
`show_output()` is a misleading name for a function that writes the SARIF file to be uploaded.

Reply:
Change it to `generate_sarif_report`